### PR TITLE
monitoring: add panels for permissions

### LINKED ISSUE #9923: Migrate Grafana Permissions panel to the new format
https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/tree/monitoring

Close #12673, #12676 along with it.

### LINKED ISSUE #12673: Site admins will not receive critical alerts if permission syncing is failing
There is a [Permissions dashboard in Grafana](https://sourcegraph.com/-/debug/grafana/d/S5j5p7gMk/permissions?orgId=1&refresh=30s) which includes a **sync error rate** panel:

image

There is no **critical** alert for this, meaning that if permissions are failing to sync 100% of the time site admins would not be notified and Sourcegraph would be unusable.

https://app.hubspot.com/contacts/2762526/company/407948923/

### LINKED ISSUE #12676: No alerts fire if repository permissions add substantial performance issues
From what I understand, authorization has some overhead added to requests. This means that if something is wrong when a site admin turns on repository permissions, request latency may go up substantially. There is a panel which shows authorization duration:

image

But there is no e.g. warning alert which would indicate this being a problem to site admins.

https://app.hubspot.com/contacts/2762526/company/407948923/

### PR DESCRIPTION
**Pull request by unknwon**
Adds Grafana panels for permissions under repo-updater dashboard.

Screenshot from my dev instance:
Screen Shot 2020-09-08 at 3 59 15 PM

Fixes #9923, fixes #12673, fixes #12676

### CODE REVIEW

File: monitoring/repo_updater.go [hunk 1]
```diff
@@ -14,6 +18,94 @@ func RepoUpdater() *Container {
...
+				Title:  "Permissions",
+				Hidden: true,
+				Rows: []Row{
+					{
+						Observable{
+							Name:              "perms_syncer_perms",
+							Description:       "time gap between least and most up to date permissions",
+							Query:             `src_repoupdater_perms_syncer_perms_gap_seconds`,
+							DataMayNotExist:   true,
+							Warning:           Alert{GreaterOrEqual: 3 * 24 * 60 * 60, For: 5 * time.Minute}, // 3 days
```
Comment (bobheadxi) [inline monitoring/repo_updater.go:31]: A bit unsure about the format here 😅 While I understand that you're trying to say "3 days" here, I would still just either:

* write the raw seconds value, with a comment (and make the definition multi-line)
* add a variable in `generator.go` for some common values (ie `HourSeconds`), but I'm not sure how widely used this is

File: monitoring/repo_updater.go [hunk 2]
```diff
@@ -14,6 +18,94 @@ func RepoUpdater() *Container {
...
+							Name:            "perms_syncer_no_perms",
+							Description:     "number of entities with no permissions",
+							Query:           `src_repoupdater_perms_syncer_no_perms`,
+							DataMayNotExist: true,
+							Warning:         Alert{GreaterOrEqual: 100, For: 5 * time.Minute},
+							PanelOptions:    PanelOptions().LegendFormat("{{type}}").Unit(Number),
+							Owner:           ObservableOwnerCloud,
+							PossibleSolutions: `
+- **Enabled permissions for the first time:** Wait for few minutes and see if the number goes down.
+- **Otherwise:** Increase the API rate limit to the code host.`,
```
Comment (bobheadxi) [inline monitoring/repo_updater.go:56]: these can be indented for readability (the generator handles the whitespace), e.g. https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/monitoring/frontend.go#L158:1-163:10

File: monitoring/repo_updater.go [hunk 3]
```diff
@@ -14,6 +18,94 @@ func RepoUpdater() *Container {
...
+						},
+						Observable{
+							Name:              "perms_syncer_stale_perms",
+							Description:       "number of entities with stale permissions",
+							Query:             `src_repoupdater_perms_syncer_stale_perms`,
+							DataMayNotExist:   true,
+							Warning:           Alert{GreaterOrEqual: 100, For: 5 * time.Minute},
+							PanelOptions:      PanelOptions().LegendFormat("{{type}}").Unit(Number),
+							Owner:             ObservableOwnerCloud,
+							PossibleSolutions: "Increase the API rate limit to the code host.",
```
Comment (bobheadxi) [inline monitoring/repo_updater.go:44]: Might be helpful to have a link to relevant documentation if there is any!

File: monitoring/repo_updater.go (hunk 1 above)
Comment (keegancsmith) [reply monitoring/repo_updater.go:31]: Also this works and is likely a bit clearer `(3 * 24 * time.Hour).Seconds()`

Comment (unknwon) [reply monitoring/repo_updater.go:31]: > Also this works and is likely a bit clearer `(3 * 24 * time.Hour).Seconds()`

I like this one, thanks!

File: monitoring/repo_updater.go (hunk 3 above)
Comment (unknwon) [reply monitoring/repo_updater.go]: Added in https://github.com/sourcegraph/sourcegraph/pull/13685/commits/972bd47d5a0a625435525ee88ef027eff00826ad!

**Review by bobheadxi — APPROVED** [review]
Nice!

**Pull request merged**